Add configuration support to the test framework to pass per-test config to components

smawson opened the pull request:
This adds a general purpose mechanism for passing configuration to components, using a marker interface to indicate that a component takes configuration. Also hooked this up to the echo test so that echo instances can be created with different configuration (service and version).

istio-testing commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/istio/istio/pull/11466#" title="Author self-approved">smawson</a>*
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **nmittler**

If they are not already assigned, you can assign the PR to them by writing `/assign @nmittler` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[pkg/OWNERS](https://github.com/istio/istio/blob/release-1.1/pkg/OWNERS)**
- **[tests/integration2/OWNERS](https://github.com/istio/istio/blob/release-1.1/tests/integration2/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["nmittler"]} -->

istio-testing commented:
@smawson: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-multicluster-e2e.sh | 1f1699c2097c34df499113fd3d06a64ced1f26e0 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11466/istio-pilot-multicluster-e2e/8050) | `/test istio-pilot-multicluster-e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@smawson: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-multicluster-e2e.sh | 1f1699c2097c34df499113fd3d06a64ced1f26e0 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11466/istio-pilot-multicluster-e2e/8050) | `/test istio-pilot-multicluster-e2e`
prow/istio-integ-k8s-tests.sh | 1f1699c2097c34df499113fd3d06a64ced1f26e0 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11466/istio-integ-k8s-tests/2898) | `/test istio-integ-k8s-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

nmittler commented on pkg/test/framework/api/component/configuration.go:
```diff
@@ -17,42 +17,42 @@ package component
-	req    Requirement
-	config Configuration
+// RequirementWrapper wraps a requirement and adds a name and a configuration.
+type RequirementWrapper struct {
```
Looking at the uses of this, I wonder if it might be better if this were a builder.  Usage might be something like this:

reqA := NewRequirementBuilder().Name("bob").Component(component.PilotID).Configuration("...").Build()

smawson replied:
I played with a couple different versions, I think I like the functional version since it is shorter code. I don't expect this to grow beyond just name and config.

We could shorten even more:
reqA := Configure(Named(ids.Pilot, "bob"), myConfig)

smawson replied:
I could also add a separate "WrapRequirement" method that takes both name and config:

reqA := WrapRequirement(&ids.Pilot, "bob", myconfig)

That would simplify a bit of the code as well.  I could go back to that and not have separate Name vs. Configure methods. What do you think?

Builder still seems like overkill here :)

nmittler replied:
yeah the functional version works.

ozevren replied:
I am not sure I fully grok why you would need this (for config purposes)?

I'd expect the usage model to simply be:
```
c := component.RequireNamed(&ids.Echo, "b")
c.Configure(echo.Config{...})
```

To hold the name, we can simply make Requirements named, with default being empty string.

louiscryan replied:
It seems like there are two concerns here - configuration for a component of type X with id Y and the ability to create variations of components with distinct ids for equivalent types.

For the former I would expect to see something like

components.ConfigureGalley(context, &ids.Galley, <some strongly typed struct>)

For the latter I would expect to see

id = context.CreateAlias(&ids.Galley, "some name")

In order to customize initialization and dependency management you would see something like

galleyVariantId = context.CreateAlias(&ids.Galley, "galley-variant")
components.ConfigureGalley(context, galleyVariantId, { initFromPixieDust : true})
components.ConfigurePilot(context, &ids.Pilot, { galleyId : galleyVariantId })

Note that the parameter for pilot declares its dependency on the galley instance id which the component would then use to initialize itself appropriately.

smawson replied:
Chatted with Louis, let me take a pass at a more general model that also allows taking dependencies on dynamic variants, will ping the PR when updated with that.

nmittler commented on pkg/test/framework/runtime/api/configurable.go:
```diff
@@ -0,0 +1,22 @@
+import "istio.io/istio/pkg/test/framework/api/component"
+
+// Configurable interface is implemented by Components that accept Configuration.
+type Configurable interface {
```
IIRC the reason for this interface is that we're going to configure components automatically while we're resolving requirements. This interface will also allow the test code to attempt to manually configure a component after it's been created, which will probably not work out so well ... but I guess it's not a huge issue.

smawson replied:
Yes, that's true, we have the same issue with Close, test code could close the component itself. Let me add a comment to the interface describing that this is for the test framework to call with the config passed with the requirement, and is not meant for tests to call directly.

nmittler replied:
sgtm

nmittler commented:
@ozevren you might want to take a look as well

istio-testing commented:
@smawson: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-multicluster-e2e.sh | 1f1699c2097c34df499113fd3d06a64ced1f26e0 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11466/istio-pilot-multicluster-e2e/8050) | `/test istio-pilot-multicluster-e2e`
prow/istio-integ-k8s-tests.sh | 1f1699c2097c34df499113fd3d06a64ced1f26e0 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11466/istio-integ-k8s-tests/2898) | `/test istio-integ-k8s-tests`
prow/istio-integ-local-tests.sh | 0884fa059d4dbd7c7701b07042783583383caee0 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11466/istio-integ-local-tests/4737) | `/test istio-integ-local-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@smawson: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-integ-k8s-tests.sh | 1f1699c2097c34df499113fd3d06a64ced1f26e0 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11466/istio-integ-k8s-tests/2898) | `/test istio-integ-k8s-tests`
prow/istio-pilot-multicluster-e2e.sh | b3a0993cc016867f9d9eb9ed255927cb74ba3846 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11466/istio-pilot-multicluster-e2e/8092) | `/test istio-pilot-multicluster-e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

smawson commented:
Ping @ozevren to take a peek, would like to get this in to continue refactoring.

incfly commented:
/cc @incfly

smawson commented:
OK brand new version of this that instead uses Descriptor directly rather than creating a new wrapper. Also reused Variant for naming instances of a component since that was kinda what it was supposed to be fore.

So major changes:
- Added a Configuration field to Descriptor
- Added Key to Descriptor, replacing ID and Variant (Key just combines them)
- Fixed resolution code to look for variants when resolving an ID
- Fixed component code to look for a default factory if a variant-specific factory is not found

New version seems to work although still some weirdness I'd like to fix.

PTAL

istio-testing commented:
@smawson: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-multicluster-e2e.sh | 35ba695be98b3add763e38dcb7336a781dea61fd | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11466/istio-pilot-multicluster-e2e/8320) | `/test istio-pilot-multicluster-e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

nmittler commented on pkg/test/framework/api/component/descriptor.go:
```diff
@@ -14,37 +14,47 @@
+	Key               Key
 	IsSystemComponent bool
-	Variant           Variant
+	Configuration     Configuration
```
Just for my own understanding ... Descriptor now seems to be a pallet of options. The fields may or may not be filled out when a component is being required, so the system will have to sort out the missing pieces (e.g. `IsSystemComponent`). I assume it's doing this via the default for the component ID? If there is an existing component available, do we re-configure in that case?

smawson replied:
Yeah this was a bit hard to decide on, mostly around requires. The model in this PR is that a variant Descriptor will override the default Descriptor (with an "" variant). For Configuration and Requires the default values will be used if they are not overridden (this means you can't remove all requirements except by replacing with something like ids.Environment). I didn't do that for IsSystemComponent, it seems like that should be explicitly set. But I haven't worked through the implications with Kube environment for that yet.

Actually I just realized I don't really always pull in the default for the component ID, that's a bit ad-hoc. Let me add a test for that and verify it works correctly, will ping PR when that is in.

There is also weirdness where if the user requests a component by ID, that means any instance of that ID, not a specific variant. That is necessary when resolving requirements, so for example if you require ids.Pilot and there is a variant of pilot running, we don't want to spin up both the variant and the default version.

I think it actually works OK, if you want a specific variant you can use a Descriptor (which we can pre-create for variants like we do with the environment variants). So in practice this looks like the right model.

smawson commented on pkg/test/framework/api/component/descriptor.go:
```diff
@@ -14,37 +14,47 @@
 type Variant string
 
+// CreateDescriptor creates a descriptor with the given ID and variant.
+func NewDescriptor(id ID, variant string) *Descriptor {
```
Yes, fixed. I thought I tried that once and it failed but I just changed it and no callers broke so yeah :)

istio-testing commented:
@smawson: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-multicluster-e2e.sh | 35ba695be98b3add763e38dcb7336a781dea61fd | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11466/istio-pilot-multicluster-e2e/8320) | `/test istio-pilot-multicluster-e2e`
prow/istio-unit-tests.sh | e09a401d59e64dacc4eb232a6c804cd1e30a7ccb | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11466/istio-unit-tests/20746) | `/test istio-unit-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@smawson: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-multicluster-e2e.sh | 35ba695be98b3add763e38dcb7336a781dea61fd | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11466/istio-pilot-multicluster-e2e/8320) | `/test istio-pilot-multicluster-e2e`
prow/istio-unit-tests.sh | e09a401d59e64dacc4eb232a6c804cd1e30a7ccb | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11466/istio-unit-tests/20746) | `/test istio-unit-tests`
prow/istio-pilot-e2e-envoyv2-v1alpha3.sh | e09a401d59e64dacc4eb232a6c804cd1e30a7ccb | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11466/istio-pilot-e2e-envoyv2-v1alpha3/15203) | `/test istio-pilot-e2e-envoyv2-v1alpha3`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@smawson: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-unit-tests.sh | e09a401d59e64dacc4eb232a6c804cd1e30a7ccb | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11466/istio-unit-tests/20746) | `/test istio-unit-tests`
prow/istio-pilot-e2e-envoyv2-v1alpha3.sh | e09a401d59e64dacc4eb232a6c804cd1e30a7ccb | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11466/istio-pilot-e2e-envoyv2-v1alpha3/15203) | `/test istio-pilot-e2e-envoyv2-v1alpha3`
prow/istio-pilot-multicluster-e2e.sh | e09a401d59e64dacc4eb232a6c804cd1e30a7ccb | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11466/istio-pilot-multicluster-e2e/8367) | `/test istio-pilot-multicluster-e2e`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@smawson: PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

istio-testing commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/istio/istio/pull/11466#" title="Author self-approved">smawson</a>*
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **geeknoid**

If they are not already assigned, you can assign the PR to them by writing `/assign @geeknoid` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/istio/istio/blob/release-1.1/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["geeknoid"]} -->

istio-testing commented:
@smawson: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-e2e-envoyv2-v1alpha3.sh | e09a401d59e64dacc4eb232a6c804cd1e30a7ccb | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11466/istio-pilot-e2e-envoyv2-v1alpha3/15203) | `/test istio-pilot-e2e-envoyv2-v1alpha3`
prow/istio-pilot-multicluster-e2e.sh | e09a401d59e64dacc4eb232a6c804cd1e30a7ccb | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11466/istio-pilot-multicluster-e2e/8367) | `/test istio-pilot-multicluster-e2e`
prow/istio-unit-tests.sh | 7589354f6a9f6ed2a1d8dd1d125912fe1f919ff0 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11466/istio-unit-tests/20852) | `/test istio-unit-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@smawson: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-e2e-envoyv2-v1alpha3.sh | e09a401d59e64dacc4eb232a6c804cd1e30a7ccb | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11466/istio-pilot-e2e-envoyv2-v1alpha3/15203) | `/test istio-pilot-e2e-envoyv2-v1alpha3`
prow/istio-pilot-multicluster-e2e.sh | e09a401d59e64dacc4eb232a6c804cd1e30a7ccb | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11466/istio-pilot-multicluster-e2e/8367) | `/test istio-pilot-multicluster-e2e`
prow/istio-unit-tests.sh | 7589354f6a9f6ed2a1d8dd1d125912fe1f919ff0 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11466/istio-unit-tests/20852) | `/test istio-unit-tests`
prow/istio-integ-local-tests.sh | 7589354f6a9f6ed2a1d8dd1d125912fe1f919ff0 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11466/istio-integ-local-tests/5251) | `/test istio-integ-local-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@smawson: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-e2e-envoyv2-v1alpha3.sh | e09a401d59e64dacc4eb232a6c804cd1e30a7ccb | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11466/istio-pilot-e2e-envoyv2-v1alpha3/15203) | `/test istio-pilot-e2e-envoyv2-v1alpha3`
prow/istio-pilot-multicluster-e2e.sh | e09a401d59e64dacc4eb232a6c804cd1e30a7ccb | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11466/istio-pilot-multicluster-e2e/8367) | `/test istio-pilot-multicluster-e2e`
prow/istio-unit-tests.sh | 7589354f6a9f6ed2a1d8dd1d125912fe1f919ff0 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11466/istio-unit-tests/20852) | `/test istio-unit-tests`
prow/istio-integ-local-tests.sh | 7589354f6a9f6ed2a1d8dd1d125912fe1f919ff0 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11466/istio-integ-local-tests/5251) | `/test istio-integ-local-tests`
prow/istio-integ-k8s-tests.sh | 7589354f6a9f6ed2a1d8dd1d125912fe1f919ff0 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11466/istio-integ-k8s-tests/3463) | `/test istio-integ-k8s-tests`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@smawson: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/istio-pilot-e2e-envoyv2-v1alpha3.sh | e09a401d59e64dacc4eb232a6c804cd1e30a7ccb | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11466/istio-pilot-e2e-envoyv2-v1alpha3/15203) | `/test istio-pilot-e2e-envoyv2-v1alpha3`
prow/istio-pilot-multicluster-e2e.sh | e09a401d59e64dacc4eb232a6c804cd1e30a7ccb | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11466/istio-pilot-multicluster-e2e/8367) | `/test istio-pilot-multicluster-e2e`
prow/istio-unit-tests.sh | 7589354f6a9f6ed2a1d8dd1d125912fe1f919ff0 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11466/istio-unit-tests/20852) | `/test istio-unit-tests`
prow/istio-integ-local-tests.sh | 7589354f6a9f6ed2a1d8dd1d125912fe1f919ff0 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11466/istio-integ-local-tests/5251) | `/test istio-integ-local-tests`
prow/istio-integ-k8s-tests.sh | 7589354f6a9f6ed2a1d8dd1d125912fe1f919ff0 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11466/istio-integ-k8s-tests/3463) | `/test istio-integ-k8s-tests`
prow/istio-presubmit.sh | 7589354f6a9f6ed2a1d8dd1d125912fe1f919ff0 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pr-logs/pull/istio_istio/11466/istio-presubmit/25611) | `/test istio-presubmit`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

smawson commented:
FYI closing this to reopen equivalent on master, since this was based on 1.1 and no more 1.1 work :)

incfly commented:
@smawson 
Is https://github.com/istio/istio/pull/11688 the master branch version replacement for this one?
